### PR TITLE
Fix MLFlow Tag Name for Resumption

### PR DESCRIPTION
**Pull request by KuuCi**
https://databricks.atlassian.net/browse/GRT-2836

### CODE REVIEW

File: composer/loggers/mlflow_logger.py
Comment (dakinggg) [inline composer/loggers/mlflow_logger.py:174]: wait, we're not turning this off in this PR, right? @jerrychen109 that would just break resumption unless I'm misunderstanding something.

Comment (jerrychen109) [reply composer/loggers/mlflow_logger.py:174]: I think we had talked about removing this shortly after we made the original change to rename the MLflow tag from `composer_run_name` to `run_name`. This was leftover from the original PR that we kept just to avoid breaking resumption for ongoing runs. There should be no runs started that are tagged with `composer_run_name` for the past 2 months, so unless there are 2-month-old runs that are ongoing and need to auto-resume with an updated composer version/commit, this should be safe? I'll leave it to your discretion though depending on how strictly we need to maintain backwards compatibility here, but I would be very very surprised if this breaks an actual user worklfow.

Comment (dakinggg) [reply composer/loggers/mlflow_logger.py:174]: ahh sorry. i misread what was being removed. thanks for clarifying, fine to delete

**Review by jerrychen109 — COMMENTED** [review]
Can you also update `test_mlflow_logger.py` to unit test the behavior when the env var is set differently from the `state.run_name`? Would also be good to verify the fallback behavior if the env var isn't set. https://github.com/mosaicml/composer/blob/83cb5c14a89fc5dcb69db0b92131e61e6ff4ae8a/tests/loggers/test_mlflow_logger.py#L259-L260

File: tests/loggers/test_mlflow_logger.py
Comment (jerrychen109) [inline tests/loggers/test_mlflow_logger.py:224]: Can you also delete this test?

**Review by jerrychen109 — APPROVED** [review]
lgtm once you get the code quality/linting to pass - thanks for adding the tests and handling the mocking! (but you'll need to get @dakinggg's approval as code owner)

File: composer/loggers/mlflow_logger.py
Comment (dakinggg) [inline composer/loggers/mlflow_logger.py:151]:
```suggestion
        self.tags['run_name'] = os.environ.get('RUN_NAME', state.run_name)
```

**Review by dakinggg — APPROVED** [review]

**Pull request merged**
